src/main.rs

```diff
@@ -52,8 +53,13 @@ mod server;
 mod test;
 
 // Timeout = 1.5s (totally arbitrary).
+#[cfg(not(test))]
 const COMPILER_TIMEOUT: u64 = 1500;
 
+// Timeout for potenially very slow CPU CI boxes
+#[cfg(test)]
+const COMPILER_TIMEOUT: u64 = 3_600_000;
```

Comment:
1hr timeout per single request in testing environment seems... overly excessive. I think we could bump it up to 10s or similar here, but for test-level timeout I'd toy more with the [TEST_TIMEOUT_IN_SEC](https://github.com/rust-lang-nursery/rls/blob/master/src/test/harness.rs#L28) value.

Reply:
This is just to deal with _potentially_ slow CI environments as tests currently (and generally) won't deal with timeouts. And if you _did_ test a timeout you wouldn't do it by waiting 2 seconds. 

I haven't actually spotted this being an issue, this is just me being defensive since this PR introduces the timeouts having an effect.

I just used an arbitrarily large number because we expect this never to happen. Maybe I should have used `u64::MAX`.